### PR TITLE
Added API requests for autocomplete as a user types in the search bar and submitted queries

### PR DESCRIPTION
**Pull request by emmymarysia**
Within this branch I added the functionality to make API requests to the Foursquare API when making a search of the map and when the user is typing in the search bar to get autocomplete results. I also added functionality that places markers on the map when the user makes a search of the results that search was called. To implement this functionality, I created two AsyncTasks within the MapFragment that are executed when the user types in the search bar for autocomplete results, and when the user submits a query from the search bar. Within the doInBackground methods of these tasks I submit the query to the API, and it is in the onPostExecute method that I parse the results. So far I have tested this by making various queries from the emulator, but plan to implement more rigorous testing with Junit that checks the results returned by my API request. Attached below is a video of what making a search currently looks like on my app.

https://user-images.githubusercontent.com/26172675/176236808-0c37b5c4-3381-48a1-9555-46f974fe6b5c.mov

### CODE REVIEW

File: app/src/main/java/com/example/golocal/fragments/MapFragment.java [hunk 1]
```diff
@@ -91,4 +163,97 @@ private void updateLocationMarker() {
             map.animateCamera(cameraUpdate);
         }
     }
+
+    private class AutocompleteCall extends AsyncTask<String, Void, String> {
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/MapFragment.java:167]: Lets move these to their own files. Private classes tend to be best when they're really, super tightly coupled to the public class, to the point that they don't make sense without eachother (classic example, linked list and private linked node). An async task seems sufficiently different from the idea of a MapFragment that they should probably be moved out. Maybe even get their own directory

File: app/src/main/java/com/example/golocal/fragments/MapFragment.java [hunk 2]
```diff
@@ -91,4 +163,97 @@ private void updateLocationMarker() {
...
+        @Override
+        protected String doInBackground(String... params) {
+            Double latitude = mCurrentLocation.getLatitude();
+            Double longitude = mCurrentLocation.getLongitude();
+            String latlng = df.format(latitude) + "%2C" + df.format(longitude);
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/MapFragment.java:172]: This name is confusing, though fun to sound out

File: app/src/main/java/com/example/golocal/fragments/MapFragment.java [hunk 3]
```diff
@@ -91,4 +163,97 @@ private void updateLocationMarker() {
...
+                results = response.body().string();
+            } catch (IOException e) {
+                e.printStackTrace();
+                return null;
+            }
+            return results;
+        }
+    }
+
+    private class SearchQueryCall extends AsyncTask<String, Void, String> {
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/MapFragment.java:193]: There's a lot of common code between this and AutocompleteCall. Can we DRY it up a bit?

**Comment by katzjm** [comment]
Great test plan for search. Could you do the same for autocomplete as well?

**Review by katzjm — APPROVED** [review]

File: app/src/main/java/com/example/golocal/fragments/MapFragment.java [hunk 4]
```diff
@@ -91,4 +163,97 @@ private void updateLocationMarker() {
...
+        BitmapDescriptor defaultMarker = BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_RED);
+        JSONObject queryResponse = new JSONObject(data);
+        LatLng mapCenter = null;
+        JSONArray results = queryResponse.getJSONArray("results");
+        queryResultBusinesses.clear();
+        for (int i = 0; i < results.length(); i++) {
+            JSONObject business = results.getJSONObject(i);
+            String address = business.getJSONObject("location").getString("address");
+            String latitude = business.getJSONObject("geocodes").getJSONObject("main").getString("latitude");
+            String longitude = business.getJSONObject("geocodes").getJSONObject("main").getString("longitude");
```
Comment (Anish2607) [inline app/src/main/java/com/example/golocal/fragments/MapFragment.java:239]: can we use named constants here too?

File: app/src/main/java/com/example/golocal/fragments/MapFragment.java [hunk 5]
```diff
@@ -91,4 +163,97 @@ private void updateLocationMarker() {
...
+            LatLng markerPosition = new LatLng(Double.parseDouble(latitude), Double.parseDouble(longitude));
+            if (i == 0) {
```
Comment (Anish2607) [inline app/src/main/java/com/example/golocal/fragments/MapFragment.java:241]: why 0? either a comment or named constant is good here

**Pull request merged**